src/webots/vrml/WbMFColor.cpp

```diff
@@ -20,7 +20,17 @@ void WbMFColor::readAndAddItem(WbTokenizer *tokenizer, const QString &worldPath)
   const double r = tokenizer->nextToken()->toDouble();
   const double g = tokenizer->nextToken()->toDouble();
   const double b = tokenizer->nextToken()->toDouble();
-  mVector.append(WbRgb(r, g, b));
+  WbRgb color(r, g, b);
+  if (color.clampValuesIfNeeded())
+    tokenizer->reportError(
+      tr("Expected positive color values in range [0.0, 1.0], found [%1 %2 %3]. MFColor field item reset to [%4 %5 %6]")
+        .arg(r)
+        .arg(g)
+        .arg(b)
```

Comment:
That may help to identify the bad guy in a MFColor field:
```suggestion
      tr("Expected positive color values in range [0.0, 1.0], found [%1 %2 %3]. MFColor field item %4 reset to [%5 %6 %7]")
        .arg(r)
        .arg(g)
        .arg(b)
        .arg(mVector.size())
```